java/src/main/java/com/logicalclocks/hsfs/metadata/StatisticsApi.java

```diff
@@ -0,0 +1,153 @@
+  public static final String SORT_BY_COMMIT_TIME_DESC = "sort_by=commit_time:desc";
+  public static final String OFFSET = "offset={offset}";
+  public static final String LIMIT = "limit={limit}";
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(StatisticsApi.class);
+
+  private EntityEndpointType entityType;
+
+  public StatisticsApi(@NonNull EntityEndpointType entityType) {
+    this.entityType = entityType;
+  }
+
+  public Statistics post(FeatureGroup featureGroup, Statistics statistics) throws FeatureStoreException, IOException {
+    return post(featureGroup.getFeatureStore().getProjectId(), featureGroup.getFeatureStore().getId(),
+      featureGroup.getId(), statistics);
+  }
+
+  public Statistics post(TrainingDataset trainingDataset, Statistics statistics)
+      throws FeatureStoreException, IOException {
+    return post(trainingDataset.getFeatureStore().getProjectId(), trainingDataset.getFeatureStore().getId(),
+      trainingDataset.getId(), statistics);
+  }
+
+  private Statistics post(Integer projectId, Integer featurestoreId, Integer entityId, Statistics statistics)
+      throws FeatureStoreException, IOException {
+    HopsworksClient hopsworksClient = getInstance();
+    String pathTemplate = PROJECT_PATH + FeatureStoreApi.FEATURE_STORE_PATH + STATISTICS_PATH;
+
+    String uri = UriTemplate.fromTemplate(pathTemplate)
+        .set("projectId", projectId)
+        .set("fsId", featurestoreId)
+        .set("entityType", entityType.getValue())
+        .set("entityId", entityId)
+        .expand();
+
+    String statisticsJson = hopsworksClient.getObjectMapper().writeValueAsString(statistics);
+    HttpPost postRequest = new HttpPost(uri);
+    postRequest.setHeader(HttpHeaders.CONTENT_TYPE, "application/json");
+    postRequest.setEntity(new StringEntity(statisticsJson));
+
+    LOGGER.info("Sending metadata request: " + uri);
+    LOGGER.info(statisticsJson);
+
+    return hopsworksClient.handleRequest(postRequest, Statistics.class);
+  }
+
+  public Statistics get(FeatureGroup featureGroup, String commitTime) throws FeatureStoreException, IOException {
+    return get(featureGroup.getFeatureStore().getProjectId(), featureGroup.getFeatureStore().getId(),
+      featureGroup.getId(), commitTime);
+  }
+
+  public Statistics get(TrainingDataset trainingDataset, String commitTime) throws FeatureStoreException, IOException {
+    return get(trainingDataset.getFeatureStore().getProjectId(), trainingDataset.getFeatureStore().getId(),
+      trainingDataset.getId(), commitTime);
+  }
+
+  private Statistics get(Integer projectId, Integer featurestoreId, Integer entityId, String commitTime)
+      throws FeatureStoreException, IOException {
+    HopsworksClient hopsworksClient = getInstance();
+    String pathTemplate = PROJECT_PATH
+        + FeatureStoreApi.FEATURE_STORE_PATH
+        + STATISTICS_PATH + "?"
+        + FILTER_COMMIT_TIME_EQ + "&"
+        + CONTENT_FIELD;
```

Comment:
The UriTemplace library provides a way of passing query params. Check the FeatureGroupApi.java class

Reply:
yeah I kind of expected there is a way to add multiple but was looking at the wrong UriTemplate documentation. Fixed